girder_worker/app.py

```diff
@@ -1,12 +1,136 @@
 import girder_worker
 import traceback as tb
+import celery
 from celery import Celery, __version__
 from distutils.version import LooseVersion
 from celery.signals import (task_prerun, task_postrun,
-                            task_failure, task_success, worker_ready)
+                            task_failure, task_success,
+                            worker_ready, before_task_publish)
+from celery.result import AsyncResult
 from .utils import JobStatus
 
 
+class GirderAsyncResult(AsyncResult):
+    def __init__(self, *args, **kwargs):
+        self._job = None
+        super(GirderAsyncResult, self).__init__(*args, **kwargs)
+
+    @property
+    def job(self):
+        if self._job is None:
+            try:
+                from girder.utility.model_importer import ModelImporter
+                job_model = ModelImporter.model('job', 'jobs')
+
+                try:
+                    return list(job_model.find({'celeryTaskId': self.task_id}))[0]
+                except IndexError:
+                    return None
+
+            except ImportError:
+                # Make a rest request to get the job info
+                return None
+
+        return self._job
+
+    @job.setter
+    def job(self, value):
+        self._job = value
```

Comment:
It's the job property read-only in terms of the API? I would think it should be assigned via some internal  mechanism such as just assigning to self._job.